landlab/components/uniform_precip/generate_uniform_precip.py

```diff
@@ -158,26 +155,43 @@ def __init__(
         # parameter dictionary or the init function, we can call the
         # different methods to assign values from the Poisson distribution.
 
-        self.storm_duration = self.get_precipitation_event_duration()
-        self.interstorm_duration = self.get_interstorm_event_duration()
-        self.storm_depth = self.get_storm_depth()
+        self._storm_duration = self.get_precipitation_event_duration()
+        self._interstorm_duration = self.get_interstorm_event_duration()
+        self._storm_depth = self.get_storm_depth()
         self._elapsed_time = 0.0
 
         # Test if we got a grid. If we did, then assign it to _grid, and we
         # are able to use the at_grid field. If not, that's cool too.
         if grid is not None:
             assert isinstance(grid, ModelGrid)  # must be a grid
-            self._grid = grid
 
         # build LL fields, if a grid is supplied:
         if grid is not None:
-            self.grid.add_field("grid", "rainfall__flux", 0.0)
+            self._grid.add_field("grid", "rainfall__flux", 0.0)
             self._gridupdate = True
         else:
             self._gridupdate = False
 
         self._intensity = self.get_storm_intensity()
 
+    @property
+    def storm_duration(self):
+        """ TODO
```

Comment:
Do these *TODO*s need to be taken care of before merging this pull request?

Reply:
Yes! Now taken care of.